src/core/services/theming/colors.js

```diff
@@ -0,0 +1,129 @@
+(function () {
+    "use strict";
+
+    //module global to keep an reference to the dynamic style sheet
+    var customSheet;
+    //module global to available color pallettes
+    var colorStore     = {};
+    var defaultPalette = '';
+
+/**
+ * @ngdoc service
+ * @name $mdColors
+ * @module material.core.color
+ *
+ * @description
+ * `$mdColors` makes it possible to use the colors of any palette to be available
+ *  as CSS rules in the form of md-bg-[colorname] and md-fg-[colorname] (as foreground and background)
+ *  If you want to activate another palette you can use the $mdColor.loadPalette(palletteName) method.
+ *
+ *  the $mdColors.palettes array lists all available palettes
+ *  the $mdColors.palette property lists the current selected palette
+ *  the $mdColors.colors array lists all the available colors with hex encoded color values for the current selected palette.
+ *
+ **/
+
+
+    angular
+        .module('material.core.colors',['material.core.theming'])
+        .service('$mdColors', mdColors)
+        .config(configColors)
+        .run(loadDefaults);
+
+    configColors.$inject = ['$mdThemingProvider'];
+    function configColors($mdThemingProvider) {
+        // fetch the colors out of the theming provider
+        Object.keys($mdThemingProvider._PALETTES).forEach(parsePalette);
+
+        // I need access to the default 'default', hardcoded for now as I miss access to it.
+        defaultPalette = $mdThemingProvider._THEMES['default'].colors.primary.name;
+        return;
+
+        // clone the palette colors to the colorStore var
+        function parsePalette(paletteName) {
+            var palette = $mdThemingProvider._PALETTES[paletteName];
+            var colors  = [];
+            colorStore[paletteName] = colors;
+            Object.keys(palette).forEach(copyColors);
+            return ;
+
+            function copyColors(colorName) {
+                // use an regex to look for hex colors, ignore the rest
+                if (/#[0-9A-Fa-f]{6}|0-9A-Fa-f]{8}\b/.exec(palette[colorName])) {
+                    colors.push({color:colorName,value:palette[colorName]});
+                }
+            }
+        }
+    }
+
+    loadDefaults.$inject=['$mdColors'];
+    function loadDefaults($mdColors) {
+        // this sets the default that is stored in the config-fase into the service!
+        $mdColors.palettes = Object.keys(colorStore);
+        $mdColors.loadPalette(defaultPalette);
+    }
+
+    mdColors.$inject = ['$interpolate'];
+    function mdColors ($interpolate) {
+        // wrap all of the above up in a reusable service.
+        var service        = this;
+        service.palette    = defaultPalette;
+        service.colors     = [];
+        service.palettes   = [];
+        service.loadPalette = loadPalette;
+
+        return service;
```

Comment:
@gkalpak I prefer it to do it this way, because it is more in line with the John Papas styleguide. Also I like to explicitly return something. 
I'm not aware of any downsides/performance issues on this approach, aside from a few extra tokens in the JS.